### DISCUSSION TITLE
EAP-MS-CHAPv2 verification failed, retry (1)

### DISCUSSION THREAD

**Discussion by TomHsiung** [strongswan/strongswan#2437] [General]
The strongSwan had been working fine. Yesterday I updated my Ubuntu server (not uprading to 24.04 TLS) and today I found cannot connect my strongSwan service by roadwarrior connects. The log shows a error,

> EAP-MS-CHAPv2 verification failed, retry (1)

All roadwarrior connects have this error log.

Today, I upgraded my Ubuntu server to 24.04 TLS. The above error remains. I don't know if the EAP-MS-CHAPv2 plugin were removed or abandoned so I cannot authorize by this method.

Looking forward to hearing from you soon! Thanks a bunch!

**Comment by tobiasbrunner** [comment]
The _eap-mschapv2_ plugin is definitely loaded and used as that error is generated by the plugin. Are there other errors in the log? (Just add the complete log.) If not, the client's password is probably wrong (i.e. the generated hashes don't match).

**Comment by TomHsiung** [comment]
> I don't know if the EAP-MS-CHAPv2 plugin were removed or abandoned so I cannot authorize by this method.

It is not true. I have tested. The problem I have is caused by some special characters in the passwords saved in the file

> /etc/ipsec.secrets

After replacing these special characters, the roadwarrior connects get back to normal working. Therefore, what changes in the Ubuntu system caused this problem? It had been working fine with these characters before yesterday's update (within 22.04 TLS).

Really want to figure it out.

**Comment by TomHsiung** [comment]
I think I made a mistake.

> I had changed the passwords for roadwarrior connects without restarting the ipsec service. In addiition, I forgot to update the Apple profiles of VPN services for my Apple clients. Therefore, it caused this error after I rebooted the server post minor upgrading.

Many thanks. I will close this discussion now.
